index.js

```diff
@@ -0,0 +1,233 @@
+// задача с самым часто встречаемым числом
+
+const arr1 = [3, 5, 3, 5, 3, 10, 3, 100, 100, 100, 100, 100, 3];
+
+const getFrequencyNumber = (arr) => {
+  const obj = arr.reduce((result, num) => {
+    if (!result[num]) {
+      result[num] = 1;
+    } else {
+      result[num] = result[num] + 1;
+    }
+    return result;
+  }, {});
+  const arrOfFrequency = [];
+  for (const key in obj) {
+    arrOfFrequency.push(obj[key]);
+  }
+  arrOfFrequency.sort((a, b) => b - a);
+
+  const maxFrequency = arrOfFrequency[0];
+  const arrOfFrequencyNumbers = [];
+  for (const key in obj) {
+    if (obj[key] === maxFrequency) {
+      arrOfFrequencyNumbers.push(key);
+    }
+  }
+  return arrOfFrequencyNumbers;
+};
+
+console.log(getFrequencyNumber(arr1));
+
+//1
+
+const users = [
+  {
+    id: 1,
+    first_name: "Jeanette",
+    last_name: "Penddreth",
+    email: "jpenddreth0@census.gov",
+    gender: "female",
+    ip_address: "26.58.193.2",
+  },
+  {
+    id: 2,
+    first_name: "Petr",
+    last_name: "Jackson",
+    email: "gfrediani1@senate.gov",
+    gender: "male",
+    ip_address: "229.179.4.212",
+  },
+];
+
+const sortByGenders = (users) => {
+  return users.reduce((result, user) => {
+    const userGender = user.gender;
+    const userName = user.first_name;
+    const userSurname = user.last_name;
+
+    user.fullName = `${userName}${userSurname}`;
+    const { first_name, last_name, ...rest } = user;
+    const modifiedUser = rest;
+
+    if (!result[userGender]) {
+      result[userGender] = [modifiedUser];
+    } else {
+      result[userGender].push(modifiedUser);
+    }
+    return result;
+  }, {});
+};
+
+console.log(sortByGenders(users));
+
+//2 Преобразуйте массив в объект используя функцию reduce.
+
+const videos = [
+  {
+    id: 65432445,
+    title: "The Chamber",
+  },
+  {
+    id: 675465,
+    title: "Fracture",
+  },
+  {
+    id: 70111470,
+    title: "Die Hard",
+  },
+  {
+    id: 654356453,
+    title: "Bad Boys",
+  },
+];
+
+const turnArrayIntoObj = (arr) => {
+  return arr.reduce((result, obj) => {
+    const id = obj.id;
+    const title = obj.title;
```

Comment:
`const { id, title } = obj`  ?) or you can destructere it like `arr.reduce((result, { id, title }) => {})`